autoemulate/core/metrics.py

```diff
@@ -0,0 +1,154 @@
+"""Metrics configuration and utilities for model evaluation and tuning."""
+
+from __future__ import annotations
+
+from collections.abc import Sequence
+from functools import partial
+
+import torchmetrics
+
+from autoemulate.core.types import MetricLike
+
+
+class Metric:
+    """Configuration for a single metric.
+
+    Parameters
+    ----------
+    metric : MetricLike
+        The torchmetrics metric class or partial.
+    name : str
+        Display name for the metric.
+    maximize : bool
+        Whether higher values are better. Defaults to True.
+    """
+
+    metric: MetricLike
+    name: str
+    maximize: bool
+
+    def __repr__(self) -> str:
+        """Return the string representation of the MetricConfig."""
+        return f"MetricConfig(name={self.name}, maximize={self.maximize})"
+
+
+class TorchMetrics(Metric):
+    """Configuration for a single torchmetrics metric.
+
+    Parameters
+    ----------
+    metric : MetricLike
+        The torchmetrics metric class or partial.
+    name : str
+        Display name for the metric. If None, uses the class name of the metric.
+    maximize : bool
+        Whether higher values are better.
+    """
+
+    def __init__(
+        self,
+        metric: MetricLike,
+        name: str,
+        maximize: bool,
+    ):
+        self.metric = metric
+        self.name = name
+        self.maximize = maximize
+
+
```

Comment:
Related to https://github.com/alan-turing-institute/autoemulate/pull/883/files#r2444396838, this could then be implemented for the subclass moving parts of model_selection module into the class, e.g.
```suggestion
    def __call__(self, y_pred: OutputLike, y_true: TensorLike) -> TensorLike:
        """Calculate metric."""
        if not isinstance(y_pred, TensorLike):
            raise ValueError(f"Metric not implemented for y_pred ({type(y_pred)})")
        if not isinstance(y_true, TensorLike):
            raise ValueError(f"Metric not implemented for y_true ({type(y_true)})")

        metric = self.metric()
        metric.to(y_pred.device)
        # Assume first dim is a batch dim, flatten others for metric calculation
        metric.update(y_pred.flatten(start_dim=1), y_true.flatten(start_dim=1))
        return metric.compute()
```